### PR TITLE
Hopefully this one is the last one :)

### PR DESCRIPTION
**Pull request by yurioni**

### CODE REVIEW

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 1]
```diff
@@ -1,95 +1,229 @@
...
+        for (person <- rentalService.allPeople) {
+          printPerson(person)
+        }
+      }
   }
 
-  private def printPerson(person: Person): Unit =
-    println("Person[id = " + person.id + ", name = " + person.name + ", surname =" + person.surname + "]")
-
+  // TODO: Uncomment when ready
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Remove the comment

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 2]
```diff
@@ -1,95 +1,229 @@
...
+      }
     } else {
       println("Rental impossible")
     }
   }
 
-  private def printAllCars = {
+  private def rentACar(carId:Long,personId:Long) = {
+    for (car<-rentalService.allCars){
+      if (carId==car.id&&car.isFree == true){
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: This expression could use some spaces for readability

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 3]
```diff
@@ -1,95 +1,229 @@
...
+        for (person<-rentalService.allPeople){
+          if (personId==person.id){
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Missing spaces around "=="

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 4]
```diff
@@ -1,95 +1,229 @@
...
+
+  private def printAllCars(withNumbers:Boolean = false) = {
+    var i = 0
+    if (withNumbers == true) {
+      for (car <- rentalService.allCars) {
+        print(i+") ")
+        printCar(car)
+        i+=1
+      }
+      }else{
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Add a space before "else"

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 5]
```diff
@@ -1,95 +1,229 @@
...
+        i += 1
+        noOccupiedCars = false
+      }
     }
+    return noOccupiedCars
   }
 
-  private def printCar(car: Car): Unit =
-    println("Car[id = " + car.id + ", model = " + car.model + ", brand = " + car.brand + ", owner = " + car.owner + "]")
+  // TODO: Uncomment when ready
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Remove the comment

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 6]
```diff
@@ -1,95 +1,229 @@
...
+      }
+    }
+  }
+
+  private def printAllOccupiedCars(withNumbers:Boolean = false):Boolean = {
+    var i = 0
+    var noOccupiedCars = true
     for (car <- rentalService.allCars) {
-      printCar(car)
+      if (car.isFree == false) {
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: equivalent to: "!car.isFree"

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 7]
```diff
@@ -1,95 +1,229 @@
...
     }
+    return noOccupiedCars
   }
 
-  private def printCar(car: Car): Unit =
-    println("Car[id = " + car.id + ", model = " + car.model + ", brand = " + car.brand + ", owner = " + car.owner + "]")
+  // TODO: Uncomment when ready
+  private def printCar(car: Car): Unit = {
+    print("Car[id = " + car.id + ", model = " + car.model + ", brand = " + car.brand + ", owner = " + car.owner + ", isFree: ")
+    if(car.isFree == true){
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: equivalent to "if(car.isFree)"

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 8]
```diff
@@ -1,95 +1,229 @@
...
 
   private def readAndSaveCar = {
-    println("Specify brand: ")
-    val brand = Console.readLine
-    println("Specify model: ")
-    val model = Console.readLine
+    var empty:Boolean = true
+    var brand:String = ""
+    var model:String = ""
+    while(empty==true) {
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Needelssly convoluted condition, use "while(empty)"

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 9]
```diff
@@ -1,95 +1,229 @@
...
+      println("Specify brand: ")
+      brand = new Scanner(System.in).nextLine()
+      empty = checkIfEmpty(brand)
+    }
+    empty = true
+    while (empty==true) {
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Same here

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 10]
```diff
@@ -1,95 +1,229 @@
...
     }
   }
 
-  private def printAllPeople = {
-    for (person <- rentalService.allPeople) {
-      printPerson(person)
-    }
+  private def printAllPeople(withNumbers:Boolean=false) = {
+    var i = 0
+    if (withNumbers == true) {
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Use "if(withNumbers)"

File: src/main/scala/prv/saevel/scala/oop/Application.scala [hunk 11]
```diff
@@ -1,95 +1,229 @@
...
+      for (person <- rentalService.allPeople) {
+        print(i + ") ")
+        printPerson(person)
+        i += 1
+      }
+    }else
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/Application.scala]: Add space before "else"

File: src/main/scala/prv/saevel/scala/oop/cars/RentalService.scala [hunk 12]
```diff
@@ -6,8 +6,16 @@ class RentalService(implicit carRepository: CarRepository, personRepository: Per
 
   def rentalPossible(carId: Long, personId: Long): Boolean = {
     val carOption = carRepository.findById(carId)
+    var isFree:Boolean = true
+      for(car<-carOption){
+      if (car.isFree ==true){
```
Comment (Saevel) [inline src/main/scala/prv/saevel/scala/oop/cars/RentalService.scala]: Equivalent to isFree = car.isFree

File: src/main/scala/prv/saevel/scala/oop/cars/RentalService.scala [hunk 13]
```diff
@@ -8,10 +8,10 @@ class RentalService(implicit carRepository: CarRepository, personRepository: Per
     val carOption = carRepository.findById(carId)
     var isFree:Boolean = true
       for(car<-carOption){
-      if (car.isFree ==true){
-        isFree =true
-      }else{
-        isFree =false
+      if (car.isFree){
```
Comment (yurioni) [inline src/main/scala/prv/saevel/scala/oop/cars/RentalService.scala:11]: @Saevel is there more "elegant" and/or direct way of accessing data stored in object to which reference we have or you have to iterate through all available ones?

**Review by Saevel — APPROVED** [review]

**Pull request merged**